### PR TITLE
Avoid unnecessary boxing/unboxing of primitives

### PR DESCRIPTION
**Pull request by stsypanov**
Wrapping primitives is pointless for the cases primitive can be used (e.g. in concatenation of arithmetics).

### CODE REVIEW

File: spring-core/src/main/java/org/springframework/cglib/proxy/Enhancer.java [hunk 1]
```diff
@@ -1254,11 +1254,11 @@ public ClassLoader getClassLoader() {
 			}
 
 			public int getOriginalModifiers(MethodInfo method) {
-				return ((Integer) originalModifiers.get(method)).intValue();
+				return (Integer) originalModifiers.get(method);
```
Comment (sbrannen) [inline spring-core/src/main/java/org/springframework/cglib/proxy/Enhancer.java:1257]: Please refrain from modifying classes under `org.springframework.asm`, `org.springframework.cglib`, and `org.springframework.objenesis`. Those include repackaged forks of the third-party libraries ASM, CGLIB, and Objenesis. Any refactoring to those classes should take place upstream in the originating repository. The Spring Framework will then pick up the changes when syncing with official updates of the forked third-party libraries.

```suggestion
				return ((Integer) originalModifiers.get(method)).intValue();
```

File: spring-core/src/main/java/org/springframework/cglib/proxy/Enhancer.java [hunk 2]
```diff
@@ -1254,11 +1254,11 @@ public ClassLoader getClassLoader() {
...
 			}
 
 			public int getIndex(MethodInfo method) {
-				return ((Integer) indexes.get(method)).intValue();
+				return (Integer) indexes.get(method);
```
Comment (sbrannen) [inline spring-core/src/main/java/org/springframework/cglib/proxy/Enhancer.java:1261]: ditto

```suggestion
				return ((Integer) indexes.get(method)).intValue();
```

File: spring-web/src/main/java/org/springframework/http/converter/ResourceRegionHttpMessageConverter.java [hunk 3]
```diff
@@ -201,10 +201,10 @@ private void writeResourceRegionCollection(Collection<ResourceRegion> resourceRe
 				print(out, "--" + boundaryString);
 				println(out);
 				if (contentType != null) {
-					print(out, "Content-Type: " + contentType.toString());
+					print(out, "Content-Type: " + contentType);
 					println(out);
 				}
-				Long resourceLength = region.getResource().contentLength();
+				long resourceLength = region.getResource().contentLength();
```
Comment (sbrannen) [inline spring-web/src/main/java/org/springframework/http/converter/ResourceRegionHttpMessageConverter.java:207]: Good catch.

That one was obviously just an accident.

**Review by sbrannen — CHANGES_REQUESTED** [review]

**Comment by sbrannen** [comment]
This has been merged into `master`.

Thanks

File: spring-web/src/main/java/org/springframework/http/converter/ResourceRegionHttpMessageConverter.java (hunk 3 above)
Comment (stsypanov) [reply spring-web/src/main/java/org/springframework/http/converter/ResourceRegionHttpMessageConverter.java:207]: No more commits into CGLIB )

**Pull request closed without merging**